privaterelay/management/commands/aggregate_generated_for.py

```diff
@@ -0,0 +1,129 @@
+from argparse import ArgumentParser
+from csv import DictReader, DictWriter
+from pathlib import Path
+from typing import Any
+from urllib.parse import urlparse
+
+from django.core.management.base import BaseCommand, CommandError
+
+
+def normalize(url: str) -> str:
+    """
+    The url in data may not have // which urlparse requires to recognize the netloc
+    script from:
+    https://stackoverflow.com/questions/53816559/python-3-netloc-value-in-urllib-parse-is-empty-if-url-doesnt-have
+    """
+    if not (
+        url.startswith("//") or url.startswith("http://") or url.startswith("https://")
+    ):
+        return "//" + url
+    return url
+
+
+def aggregate_by_generated_for(file_path: str) -> Path:
+    aggregate_usage: dict[str, dict[str, int]] = {}
+    with open(file_path, newline="") as csvfile:
+        datareader = DictReader(csvfile, delimiter=",", quotechar="|")
+        columns = [
+            "count",  # Number of masks with the generated_for
+            "total_usage",  # Sum of emails forwarded, emails blocked,
+            # trackers blocked in emails, emails replied, and spam blocked
+            "total_forwarded",  # Total emails forwarded to masks
+            "total_blocked",  # Total emails blocked for masks
+            "total_level_one_trackers_blocked",  # Total number of trackers
+            # blocked in emails forwarded to masks
+            "total_replied",  # Total number of emails replied to masks
+            "total_spam",  # Total number of spam
+        ]
+
+        for row in datareader:
+            aggregate_data: dict[str, int] = {
+                "count": 0,
+                "row_count": 0,
+                "total_usage": 0,
+                "total_forwarded": 0,
+                "total_blocked": 0,
+                "total_level_one_trackers_blocked": 0,
+                "total_replied": 0,
+                "total_spam": 0,
+            }
+            url = row["generated_for"]
+
+            # TODO: clean the domain for multiple domains in generated_for
+            # separated by space, strip www, and others like stripping path
+            if url:
+                normalized_url = normalize(url)
+                domain = urlparse(normalized_url).netloc
+            else:
+                domain = url
+
+            if domain in aggregate_usage:
+                aggregate_data = aggregate_usage[domain]
+
+            aggregate_data["row_count"] = aggregate_data["row_count"] + 1
+            for col in columns:
+                d = 0
+                try:
+                    d = int(row[col])
+
+                except ValueError as e:
+                    print(e)
+                    print(f"{col}: {row[col]}")
```

Comment:
Did this happen when you processed the data? Or was this debug code that stayed in?

Reply:
It did but only because of the way I split the row using commas. There were multiple values in `generated_for` or `used_on` that was separated by `,` which resulted in this error. It's now fixed by recommending that the comma is replaced with space in the `Process Data` step in the "Relay Quarterly Mask Acceptance Testing Data" doc. I think it's safe to remove.

Reply:
Removed